evap/results/views.py

```diff
@@ -135,13 +136,26 @@ def index(request):
         additional_evaluations = get_evaluations_with_course_result_attributes(additional_evaluations)
         evaluations += additional_evaluations
 
-    evaluations.sort(key=lambda evaluation: (evaluation.course.semester.pk, evaluation.full_name))  # evaluations must be sorted for regrouping them in the template
+    # put evaluations into a dict that maps from course to a list of evaluations.
+    # this dict is sorted by course.pk (important for the zip below)
+    # (this relies on python 3.7's guarantee that the insertion order of the dict is preserved)
+    evaluations.sort(key=lambda evaluation: evaluation.course.pk)
 
-    evaluation_pks = [evaluation.pk for evaluation in evaluations]
-    degrees = Degree.objects.filter(courses__evaluations__pk__in=evaluation_pks).distinct()
-    course_types = CourseType.objects.filter(courses__evaluations__pk__in=evaluation_pks).distinct()
+    courses = defaultdict(list)
+    for evaluation in evaluations:
+        courses[evaluation.course].append(evaluation)
```

Comment:
Have you thought about doing something like ```
evaluation.course.num_evaluations = evaluation.course.num_evaluations + 1 if evaluation.course.num_evaluations else 0 ```
here instead of the query logic below?

Reply:
I don't quite understand, where does `num_evaluations` come from if not from the query logic?

Reply:
We can simply count up the count of evaluations from 0 in this loop, it runs once for every evaluation anyway.

Doing it that way instead of querying the database for the number of evaluations would save that query -  but I'm not sure about whether it's better.

Reply:
`evaluations` does not contain all evaluations, but is filtered. here i actually need all evaluations. should i add a comment in that regard?

Reply:
Ah, I see. No, I think it's fine, I should have just read more context.